README.md

```diff
@@ -219,91 +211,141 @@ import java.sql.*;
+
+By default, Enhanced Failure Monitoring is loaded. Additional custom plugins can be implemented and used alongside existing ones. Plugins can be chained together in a desired order. Loading custom plugins will not include Enhanced Failure Monitoring unless explicitly stated through the `connectionPluginFactories` parameter.
+
+The AWS JDBC Driver for MySQL attaches the `DefaultConnectionPlugin` to the tail of the connection plugin chain 
+and actually executes the given JDBC method.
+
+Since all the connection plugins are chained together, the prior connection plugin affects the
+latter plugins. If the connection plugin at the head of the connection plugin chain measures the
+execution time, this measurement would encompass the time spent in all the connection plugins down
+the chain.
+
+To learn how to write custom plugins, refer to examples located inside [Custom Plugins Demo](https://github.com/awslabs/aws-mysql-jdbc/tree/main/src/demo/java/customplugins).
+
+#### Connection Plugin Manager Parameters
+| Parameter       | Value           | Required      | Description  | Default Value |
+| ------------- |:-------------:|:-------------:|:-------------:| --------- |
+|`connectionPluginFactories` | String | No | String of fully-qualified class name of plugin factories. <br/><br/>Each factory in the string should be comma-separated `,`<br/><br/>**NOTE: The order of factories declared matters.**  <br/><br/>Example: `customplugins.MethodCountConnectionPluginFactory`, `customplugins.MethodCountConnectionPluginFactory,customplugins.ExecutionTimeConnectionPluginFactory,com.mysql.cj.jdbc.ha.ca.plugins.NodeMonitoringConnectionPluginFactory` | `com.mysql.cj.jdbc.ha.ca.plugins.NodeMonitoringConnectionPluginFactory` |
+
+>### :warning: Warnings About Connection Plugin Manager
+>If application is unable to close, it may be due to connection plugin manager not being able to release resources properly. User can manually release resources by calling `software.aws.rds.jdbc.mysql.Driver.releasePluginManagers()`.
+
+### Enhanced Failure Monitoring
+<div style="text-align:center"><img src="./docs/files/images/enhanced_failure_monitoring_diagram.png" /></div>
+The figure above shows a simplified workflow of enhanced failure monitoring. Enhanced failure monitoring, is a connection plugin implemented by using a monitor thread. The monitor will periodically check the connected database node's health. In the case of the database node showing up as unhealthy, the query will be retried with a new database node and the monitor is restarted. 
+
+Enhanced Failure Monitoring is loaded in by default and can be disabled by setting parameter `failureDetectionEnabled` to `false`. 
```

Comment:
We sometimes use `Enhanced Failure Monitoring` and sometimes use `enhanced failure monitoring`. We should be more consistent, do we want to consider this a proper noun or not?